snuba/query/processors/match_replace_processor.py

```diff
@@ -0,0 +1,33 @@
+from dataclasses import replace
+from typing import Callable
+
+from snuba.query.expressions import Expression
+from snuba.query.logical import Query
+from snuba.query.matchers import Pattern, MatchResult
+from snuba.query.processors import QueryProcessor
```

Comment:
Pay attention, this is only a dataset query processor. Are you planning to build both the logical query processor and the physical query processor or is the logical one the only needed one ?

Reply:
Right now I think the logical one is the only needed one. At least it's the only one I have a use case for at the moment.